create_jenkins_job.py

```diff
@@ -123,6 +123,8 @@ def main(argv=None):
             'ignore_rmw_default': data['ignore_rmw_default'] | {'rmw_connext_cpp', 'rmw_connext_dynamic_cpp'},
             'build_args_default': data['build_args_default'].replace(
                 '--cmake-args', '--cmake-args -DCMAKE_CXX_FLAGS=-Wno-psabi -DCMAKE_C_FLAGS=-Wno-psabi -DDISABLE_SANITIZERS=ON'),
+            # bionic apt python3-pygraphviz (1.4~rc1) is broken on armhf, disabling only the tests since the package is only used by the tests
```

Comment:
Is it really the case that this is only used in tests?  https://github.com/ros-visualization/qt_gui_core/blob/kinetic-devel/qt_dotgraph/src/qt_dotgraph/pygraphvizfactory.py#L33 suggests otherwise.  If that's the case, then we'll probably have to do something different here, since we don't want to package (for instance) something that can't possibly work.  Can you elaborate more on how python3-pygraphviz is broken on Bionic?

Reply:
Within this package, pygraphvizfactory is only imported by the tests, and the comment below what you posted says it's a "reference implementation" - but since it's part of the exposed API there is the possibility it's being used.

The issue is - using the following test file, and running `python3 test.py`, the program exits with no output as expected on x86_64 and aarch64

```
# test.py
#!/usr/bin/env python3

import pygraphviz

graph = pygraphviz.AGraph(directed=True, ranksep=0.2, rankdir='TB', rank='same', simplify=True)
graph.add_node('foo')
graph.add_node('bar')
graph.add_edge('foo', 'bar')
```


However, on arm32 (tried two ways - qemu on an x86_64 machine through Docker, and 32 bit Docker on an aarch64 EC2 instance), the following error happens

```
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/pygraphviz/agraph.py", line 1561, in __new__
    nh = gv.agnode(graph.handle, n.encode(graph.encoding), _Action.find)
KeyError: 'agnode: no key'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "test.py", line 8, in <module>
    graph.add_edge('foo', 'bar')
  File "/usr/lib/python3/dist-packages/pygraphviz/agraph.py", line 478, in add_edge
    e = Edge(self, eh=eh)
  File "/usr/lib/python3/dist-packages/pygraphviz/agraph.py", line 1625, in __new__
    s = Node(graph, nh=source)
  File "/usr/lib/python3/dist-packages/pygraphviz/agraph.py", line 1563, in __new__
    raise KeyError("Node %s not in graph." % n)
KeyError: 'Node None not in graph.'
```

If I upgrade to 1.5 via pip after installing the apt dependency `libgraphviz-dev`, this same snippet works on all of these platforms.

Maybe disabling the test is not the right idea, perhaps the dependency needs to be changed to use the latest from pip instead - but this process seems to always be questionable for the Windows build.

Reply:
I see what you mean about it working with pygraphviz 1.5 from pip, but the really weird thing is that it *also* works just fine if I install pygraphviz 1.4rc1 from pip (it does not work when installing from the debian package).  Also, it seems to work with the debian packages python3-pygraphviz 1.5 from disco, and python3-pygraphviz 1.4(!) from cosmic.  As far as I can tell, there are no differences between the cosmic and the bionic packages, other than it was rebuilt for Python 3.7.

Here's a summary:

| OS | pygraphviz version | Installed from debian or pip | Example Works? |
| ---- | ------------------------ | --------------------------------- | ---------- |
| Bionic | 1.4~rc1 | Debian | No |
| Cosmic | 1.4~rc1 | Debian | Yes |
| Disco | 1.5 | Debian | Yes |
| Bionic | 1.4~rc1 | pip | Yes |
| Bionic | 1.5 | pip | Yes |

And one more data point; if I pull down the debian sources (`apt-get source python3-pygraphviz ; apt-get build-dep python3-pygraphviz`), build with no modifications (`debuild -uc -us`), and then install (`dpkg -i python3-pygraphviz_1.4~rc1-1build2_armhf.deb`), it works just fine.

From all of this, I can only conclude that the Ubuntu Bionic debian package is somehow corrupt.  It may be worthwhile following up with Canonical about this, and seeing if they are willing to do a rebuild of the package; that would solve all of the issues here.  There are various things we can do to workaround it, but it would be better for everyone if we got this fixed in upstream Bionic.  @emersonknapp would you mind trying to do that?

Reply:
I'm on it - thanks for the extra tests, that's really informative

Reply:
Once you've found or opened a bug report with Ubuntu please link it here.

Reply:
See main thread of this issue :)

Reply:
> See main thread of this PR :)
Ah thanks. I had the tab open already and the dynamic update failed :bow: